File: javaclient/javaclient/specs/SpecsRequestBuilder.java
package javaclient.specs;

import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import java.util.HashMap;
import java.util.Objects;
import javaclient.specs.item.WithProviderItemRequestBuilder;
/**
 * Builds and executes requests for operations under /specs
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class SpecsRequestBuilder extends BaseRequestBuilder {
    /**
     * Gets an item from the javaclient.specs.item collection
     * @param provider Unique identifier of the item
     * @return a {@link WithProviderItemRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public WithProviderItemRequestBuilder byProvider(@jakarta.annotation.Nonnull final String provider) {
        Objects.requireNonNull(provider);
        final HashMap<String, Object> urlTplParams = new HashMap<String, Object>(this.pathParameters);
        urlTplParams.put("provider", provider);
        return new WithProviderItemRequestBuilder(urlTplParams, requestAdapter);
    }
    /**
     * Instantiates a new {@link SpecsRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public SpecsRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/specs", pathParameters);
    }
    /**
     * Instantiates a new {@link SpecsRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public SpecsRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/specs", rawUrl);
    }
}

File: goclient/goclient/specs/specs_request_builder.go
package specs

import (
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f "github.com/microsoft/kiota-abstractions-go"
)

// SpecsRequestBuilder builds and executes requests for operations under \specs
type SpecsRequestBuilder struct {
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.BaseRequestBuilder
}
// ByProvider gets an item from the github.com/andrueastman/UntypedNodeSamples/GoClient/GoClient.specs.item collection
// returns a *WithProviderItemRequestBuilder when successful
func (m *SpecsRequestBuilder) ByProvider(provider string)(*WithProviderItemRequestBuilder) {
    urlTplParams := make(map[string]string)
    for idx, item := range m.BaseRequestBuilder.PathParameters {
        urlTplParams[idx] = item
    }
    if provider != "" {
        urlTplParams["provider"] = provider
    }
    return NewWithProviderItemRequestBuilderInternal(urlTplParams, m.BaseRequestBuilder.RequestAdapter)
}
// NewSpecsRequestBuilderInternal instantiates a new SpecsRequestBuilder and sets the default values.
func NewSpecsRequestBuilderInternal(pathParameters map[string]string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*SpecsRequestBuilder) {
    m := &SpecsRequestBuilder{
        BaseRequestBuilder: *i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.NewBaseRequestBuilder(requestAdapter, "{+baseurl}/specs", pathParameters),
    }
    return m
}
// NewSpecsRequestBuilder instantiates a new SpecsRequestBuilder and sets the default values.
func NewSpecsRequestBuilder(rawUrl string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*SpecsRequestBuilder) {
    urlParams := make(map[string]string)
    urlParams["request-raw-url"] = rawUrl
    return NewSpecsRequestBuilderInternal(urlParams, requestAdapter)
}
